pkg/controllers/managementagent/monitoring/operatorHandler.go

```diff
@@ -40,32 +42,31 @@ func (h *operatorHandler) syncCluster(key string, obj *mgmtv3.Cluster) (runtime.
 		return obj, nil
 	}
 
-	var newCluster *mgmtv3.Cluster
-	var err error
+	cpy := obj.DeepCopy()
+	var returnErr error
+	var newObj runtime.Object
 	//should deploy
 	if obj.Spec.EnableClusterAlerting || obj.Spec.EnableClusterMonitoring {
-		newObj, err := v32.ClusterConditionPrometheusOperatorDeployed.Do(obj, func() (runtime.Object, error) {
-			cpy := obj.DeepCopy()
+		newObj, returnErr = v32.ClusterConditionPrometheusOperatorDeployed.Do(cpy, func() (runtime.Object, error) {
 			return cpy, deploySystemMonitor(cpy, h.app, h.catalogManager, h.clusterName)
 		})
-		if err != nil {
-			logrus.Warnf("deploy prometheus operator error, %v", err)
-		}
-		newCluster = newObj.(*mgmtv3.Cluster)
+		cpy = newObj.(*mgmtv3.Cluster)
 	} else { // should withdraw
-		newCluster = obj.DeepCopy()
-		if err = withdrawSystemMonitor(newCluster, h.app); err != nil {
-			logrus.Warnf("withdraw prometheus operator error, %v", err)
-		}
+		returnErr = withdrawSystemMonitor(cpy, h.app)
 	}
 
-	if newCluster != nil && !reflect.DeepEqual(newCluster, obj) {
-		if newCluster, err = h.clusters.Update(newCluster); err != nil {
-			return nil, err
+	if !reflect.DeepEqual(cpy, obj) {
+		retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			_, err := h.clusters.Update(cpy)
```

Comment:
Do we really need `RetryOnConflict` to retry updating cluster? I think the `clusters.Update` already include the retry logic.

Reply:
@orangedeng There is not a retry built into `clusters.Update`, the framework for controllers handles this by enqueuing the cluster again when an error is returned which causes all handles to run on the cluster again. 

Because of all the changes being done to the cluster object inside the helper functions I agree we shouldn't use it here.